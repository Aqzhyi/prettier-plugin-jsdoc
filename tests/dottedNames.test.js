/* eslint-disable no-undef */
const prettier = require("prettier");

function subject(code, options = {}) {
  return prettier.format(code, {
    parser: "jsdoc-parser",
    plugins: ["."],
    jsdocSpaces: 1,
    ...options,
  });
}

test("dotted names function param", () => {
  const result = subject(`
  /**
   * @param {object} data
   * @param {string} data.userName
   * @param {string} data.password
   *
   * @typedef {object} LoginResponse
   * @property {string} token
   * @property {number} expires
   * @property {boolean} mustChangePassword
   *
   * @returns {import('axios').AxiosResponse<LoginResponse>}
   */
    function a(){}
`);

  expect(result).toMatchSnapshot();
});
