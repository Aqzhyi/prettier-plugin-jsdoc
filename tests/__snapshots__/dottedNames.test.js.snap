// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotted names function param 1`] = `
"/**
 * @typedef {object} LoginResponse
 * @property {string} token
 * @property {number} expires
 * @property {boolean} mustChangePassword
 * @param {object} data
 * @param {string} data.userName
 * @param {string} data.password
 * @returns {import(\\"axios\\").AxiosResponse<LoginResponse>}
 */
function a() {}
"
`;
